Allow client auth name/key to be read directly from a file

gavin-norman-sociomantic opened the pull request.

Commit:
Fix minor formatting issues

Commit:
Move node-specific credentials code into own module

The credentials file parsing code will also be needed in the client.

Part of #53.

Commit:
Fix internal deprecations introduced by credentials refactorings

Commit:
Add functions to read client credentials from a file

Part of #53.

Commit:
Add client constructor accepting path to credentials file

This simplifies usage: the user no longer has to manually provide
the auth name/key to use.

Fixes #53.

gavin-norman-sociomantic commented:
Blocked while testing with a real client.

nemanja-boric-sociomantic commented on src/swarm/neo/authentication/Credentials.d:
```diff
@@ -11,98 +11,52 @@
+*******************************************************************************/
 
-    invariant ( )
+public static size_t validateNameCharacters ( cstring name )
```
`static` is now extra

nemanja-boric-sociomantic commented on src/swarm/neo/authentication/CredentialsFile.d:
```diff
@@ -38,275 +38,219 @@
+    size_t file_length = file.length;
 
-    public this ( istring filepath )
+    if (file_length > CredDef.LengthLimit.File)
```
I feel we can work around this limit in the future - it doesn't strike me like a something that's a necessity. Not related to this PR.

gavin-norman-sociomantic replied:
Yeah, I was also surprised to find this limit. It's a simplistic implementation that we can extend at some point, if needed.

nemanja-boric-sociomantic commented:
Looks good!

gavin-norman-sociomantic commented:
Tested. Works as expected. Unblocked.

Commit:
Move client-specific credentials code into own module

Just to clarify the usage of the different elements (i.e. which are
only used in the client and which are universal).

gavin-norman-sociomantic commented:
Updated:
* Removed `static` from free function.

nemanja-boric-sociomantic commented:
LGTM
